src/backend/distributed/utils/node_metadata.c

```diff
@@ -1052,7 +1052,7 @@ GetNodeTuple(char *nodeName, int32 nodePort)
 	ScanKeyInit(&scanKey[0], Anum_pg_dist_node_nodename,
 				BTEqualStrategyNumber, F_TEXTEQ, CStringGetTextDatum(nodeName));
 	ScanKeyInit(&scanKey[1], Anum_pg_dist_node_nodeport,
-				BTEqualStrategyNumber, F_INT8EQ, Int32GetDatum(nodePort));
+				BTEqualStrategyNumber, F_INT4EQ, Int32GetDatum(nodePort));
```

Comment:
We have the exact same problem in `metadata_sync.c` -> `MarkNodeHasMetadata`. Please also fix that.

Reply:
pushed